common/src/main/java/com/wynntils/handlers/scoreboard/ScoreboardHandler.java

```diff
@@ -100,80 +93,39 @@ public void onWorldStateChange(WorldStateEvent event) {
 
         // 4. Create our own scoreboard to hide specific segments
-        createScoreboardFromSegments();
+        createScoreboardFromSegments(reconstructedScoreboard);
     }
 
-    private void handleScoreChange(StyledText owner, int score) {
-        // A score change can mean two things:
-        // 1. A new line was added to the scoreboard
-        // 2. An existing line with the same score was changed
-        // 3. An existing line's score was changed
+    private List<ScoreboardLine> getCurrentScoreboardState(String currentScoreboardName) {
+        Scoreboard scoreboard = McUtils.mc().level.getScoreboard();
+        Objective objective = scoreboard.getObjective(currentScoreboardName);
+        List<Score> lines = new ArrayList<>(scoreboard.getPlayerScores(objective));
 
-        // First, we check if the line already exists, by using the score as the identifier
-        ScoreboardLine existingScore = reconstructedScoreboard.stream()
-                .filter(line -> line.score() == score)
-                .findFirst()
-                .orElse(null);
+        // Lines are by default in reverse order
+        Collections.reverse(lines);
 
-        if (existingScore != null) {
-            // The line already exists, so we just update the line
-            reconstructedScoreboard.remove(existingScore);
-            reconstructedScoreboard.add(new ScoreboardLine(owner, score));
-            return;
-        }
-
-        // Secondly, we check if the line already exists, by using the owner as the identifier
-        ScoreboardLine existingLine = reconstructedScoreboard.stream()
-                .filter(line -> line.line().equals(owner))
-                .findFirst()
-                .orElse(null);
-
-        if (existingLine != null) {
-            // The line already exists, so we just update the score
-            reconstructedScoreboard.remove(existingLine);
-            reconstructedScoreboard.add(new ScoreboardLine(owner, score));
-            return;
-        }
-
-        // The line doesn't exist, so we add it
-        reconstructedScoreboard.add(new ScoreboardLine(owner, score));
-    }
-
-    private void handleScoreRemove(StyledText owner) {
-        // A score remove can mean two things:
-        // 1. An existing line was removed
-        // 2. The line to be removed was changed before, so it doesn't exist anymore
-
-        // First, we check if the line exists, and remove it
-        // If it doesn't, we don't need to do anything
-        reconstructedScoreboard.stream()
-                .filter(line -> line.line().equals(owner))
-                .findFirst()
-                .ifPresent(reconstructedScoreboard::remove);
+        return lines.stream()
+                .map(s -> new ScoreboardLine(StyledText.fromString(s.getOwner()), s.getScore()))
+                .toList();
     }
 
-    private boolean isScoreboardValid() {
+    private boolean isScoreboardValid(List<ScoreboardLine> reconstructedScoreboard) {
```

Comment:
I can't comment on the right lines (thanks, github :-/) but the description below has some typos and strange grammar.